tests/init/tinit_var_tracking_tuples.nim

```diff
@@ -0,0 +1,28 @@
+discard """
+  nimout: '''
+tinit_var_tracking_tuples.nim(14, 7) Hint: BEGIN [User]
+tinit_var_tracking_tuples.nim(25, 7) Hint: END [User]
+'''
+description: '''
+  . From https://github.com/nim-lang/Nim/issues/8314
+    Incorrect 'Cannot prove that 'y' is initialized' caused by tuple assignment
+  . If a normal assignment is used instead, the warning disappears.
+  . Fixed by https://github.com/nim-lang/Nim/pull/8321
+'''
+"""
+
+{.hint: "BEGIN".}
```

Comment:
Not sure why we need these hints?

Likely just need an assert to make sure there is an actual call and test

Reply:
https://github.com/nim-works/nimskull/pull/327#discussion_r902722393

Reply:
I still don't get it, but we can figure that out of band. 😆

Reply:
I removed the hint and added another doAssert based on the output of the function, just to be sure.